src/test/java/io/candydoc/ddd/extract_domain_vocabulary/GenericTypesTest.java

```diff
@@ -0,0 +1,200 @@
+package io.candydoc.ddd.extract_domain_vocabulary;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+
+import io.candydoc.ddd.model.Dependency;
+import io.candydoc.ddd.model.Description;
+import io.candydoc.ddd.model.FullName;
+import io.candydoc.ddd.model.PackageName;
+import io.candydoc.ddd.model.SimpleName;
+import io.candydoc.ddd.model.UbiquitousLanguage;
+import io.candydoc.ddd.strategic_pattern.BoundedContext;
+import io.candydoc.ddd.tactical_pattern.Aggregate;
+import io.candydoc.ddd.tactical_pattern.DomainCommand;
+import io.candydoc.ddd.tactical_pattern.DomainEvent;
+import io.candydoc.ddd.tactical_pattern.TacticalPattern;
+import io.candydoc.ddd.tactical_pattern.ValueObject;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class GenericTypesTest {
+  @Spy DomainVocabularyFinder domainVocabularyFinder = new ReflectionsDomainVocabularyFinder();
+  @Mock UbiquitousLanguageRepository ubiquitousLanguageRepository;
+  @InjectMocks ExtractDomainModelsUseCase useCase;
+  @Captor ArgumentCaptor<Set<UbiquitousLanguage>> ubiquitousLanguagesCaptor;
+
+  @Test
+  public void find_patterns_from_generic_types() {
```

Comment:
This test is huge, maybe you can find a way to split it in different examples based on the class you're looking at.

Something like:
- find_patterns_in_list
- find_patterns_in_map
- ...